File: 0017-letter-combinations-of-a-phone-number/0017-letter-combinations-of-a-phone-number.py
class Solution(object):
    def letterCombinations(self, digits):
        """
        :type digits: str
        :rtype: List[str]
        """
        #if digits is empty return empty str
        if not digits:
            return []
        else:
            arrDigit = {
                '2': "abc", '3': "def", '4': "ghi",
                '5': "jkl", '6': "mno", '7': "pqrs",
                '8': "tuv", '9': "wxyz"
            }

            arrCom = ['']

            for i in digits:
                arrTemp = []
                for x in arrDigit[i]:
                    for y in arrCom:
                        arrTemp.append(y + x)
                arrCom = arrTemp

            return arrCom
                      

File: 0017-letter-combinations-of-a-phone-number/0017-letter-combinations-of-a-phone-number.cs
public class Solution {
    public IList<string> LetterCombinations(string digits) {
        if (string.IsNullOrEmpty(digits)) return new List<string>();
        
        Dictionary<char, string> arrDigit = new Dictionary<char, string>() {
            {'2', "abc"}, {'3', "def"}, {'4', "ghi"},
            {'5', "jkl"}, {'6', "mno"}, {'7', "pqrs"},
            {'8', "tuv"}, {'9', "wxyz"}
        };

        List<string> arrCom = new List<string>() {""};

        foreach (char digit in digits) {
            List<string> arrTemp = new List<string>();
            foreach (char x in arrDigit[digit]) {
                foreach (string y in arrCom) {
                    arrTemp.Add(y + x);
                }
            }
            arrCom = arrTemp;
        }

        return arrCom;
    }
}